packages/forms-web-app/src/dynamic-forms/middleware/redirect-to-unanswered-question.js

```diff
@@ -4,21 +4,19 @@ module.exports = () => (req, res, next) => {
 	const journeyResponse = res.locals.journeyResponse;
 	const journey = getJourney(journeyResponse);
 
-	const questionFieldNames = new Set(
-		journey.sections.flatMap((section) => section.questions.map((question) => question.fieldName))
-	);
+	for (const section of journey.sections) {
+		for (const question of section.questions) {
+			const answer = journey.response?.answers[question.fieldName];
+			if (question.fieldName === 'uploadLpaStatementDocuments') {
+				const additionalDocumentsAnswer = journey.response.answers['additionalDocuments'];
+				if (additionalDocumentsAnswer !== 'yes') {
+					continue;
+				}
+			}
```

Comment:
This feels like a very specific condition to have in such a general seeming middleware. Maybe this middleware could become higher order and accept an array of functions which take question and answer and return a bool? So the middleware signature is
```
/**
 * @param {[((question: import('../section').Question, answer: unknown) => boolean)]} conditions
 * @returns {import('express').Handler}
 */
module.exports = (conditions) => (req, res, next) => {
...
```
then in this loop you run the condition functions. That way this middleware doesn't need to know about whatever journey `uploadLpaStatementDocuments` is in reference to _and_ we can use it for other questions later if we need to

Reply:
@Si3rr4wow Have just pushed changes, something like this?

Reply:
Yeah nice one!